diff --git a/src/java.desktop/unix/classes/sun/font/FcFontConfiguration.java b/src/java.desktop/unix/classes/sun/font/FcFontConfiguration.java
index 9808e46ee1f..48de9466281 100644
--- a/src/java.desktop/unix/classes/sun/font/FcFontConfiguration.java
+++ b/src/java.desktop/unix/classes/sun/font/FcFontConfiguration.java
@@ -352,7 +352,10 @@ public class FcFontConfiguration extends FontConfiguration {
             String userDir = System.getProperty("user.home");
             String version = System.getProperty("java.version");
             String fs = File.separator;
-            String dir = userDir+fs+".java"+fs+"fonts"+fs+version;
+            String xdgCacheDir = System.getenv("XDG_CACHE_HOME");
+            if (xdgCacheDir == null)
+                xdgCacheDir = userDir+fs+".cache";
+            String dir = xdgCacheDir+fs+"java"+fs+"fonts"+fs+version;
             Locale locale = SunToolkit.getStartupLocale();
             String lang = locale.getLanguage();
             String country = locale.getCountry();
diff --git a/src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java b/src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java
index b30a23465c5..1757ab9ecd6 100644
--- a/src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java
+++ b/src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java
@@ -114,9 +114,15 @@ class FileSystemPreferences extends AbstractPreferences {
     private static void setupUserRoot() {
         AccessController.doPrivileged(new PrivilegedAction<Void>() {
             public Void run() {
-                userRootDir =
-                      new File(System.getProperty("java.util.prefs.userRoot",
-                      System.getProperty("user.home")), ".java/.userPrefs");
+                String preferredUserRoot = System.getProperty("java.util.prefs.userRoot");
+                String userHome = System.getProperty("user.home");
+                String xdgConfigHome = System.getenv("XDG_CONFIG_HOME");
+                if (xdgConfigHome == null)
+                    xdgConfigHome = userHome + File.separator + ".config";
+                if (preferredUserRoot == null)
+                    userRootDir = new File(xdgConfigHome, "java" + File.separator + "userPrefs");
+                else
+                    userRootDir = new File(preferredUserRoot, ".java/.userPrefs");
                 // Attempt to create root dir if it does not yet exist.
                 if (!userRootDir.exists()) {
                     if (userRootDir.mkdirs()) {
