--- a/bin/idea.sh	1970-01-20 12:40:00.000000000 +0200
+++ b/bin/idea.sh	2023-03-29 09:58:13.803845123 +0300
@@ -8,17 +8,7 @@
 message()
 {
   TITLE="Cannot start IntelliJ IDEA"
-  if [ -n "$(command -v zenity)" ]; then
-    zenity --error --title="$TITLE" --text="$1" --no-wrap
-  elif [ -n "$(command -v kdialog)" ]; then
-    kdialog --error "$1" --title "$TITLE"
-  elif [ -n "$(command -v notify-send)" ]; then
-    notify-send "ERROR: $TITLE" "$1"
-  elif [ -n "$(command -v xmessage)" ]; then
-    xmessage -center "ERROR: $TITLE: $1"
-  else
-    printf "ERROR: %s\n%s\n" "$TITLE" "$1"
-  fi
+  @NOTIFY_SEND@ "ERROR: $TITLE" "$1"
 }
 
 if [ -z "$(command -v uname)" ] || [ -z "$(command -v realpath)" ] || [ -z "$(command -v dirname)" ] || [ -z "$(command -v cat)" ] || \
@@ -31,6 +21,9 @@
   exit 1
 fi
 
+PATH=$PATH:'@PATH@'
+NATIVE_LIBRARY_PATH='@NATIVE_LIBRARY_PATH@'
+
 # shellcheck disable=SC2034
 GREP_OPTIONS=''
 OS_TYPE=$(uname -s)
@@ -168,6 +161,7 @@
 # shellcheck disable=SC2086
 exec "$JAVA_BIN" \
   -classpath "$CLASS_PATH" \
+  -Djna.library.path="$NATIVE_LIBRARY_PATH" \
   "-XX:ErrorFile=$HOME/java_error_in_idea_%p.log" \
   "-XX:HeapDumpPath=$HOME/java_error_in_idea_.hprof" \
   ${VM_OPTIONS} \
