From 713481126626f8add03a2c65a733e78964d67e85 Mon Sep 17 00:00:00 2001
From: Olli Helenius <liff@iki.fi>
Date: Wed, 9 Aug 2023 21:34:16 +0300
Subject: [PATCH] =?UTF-8?q?Don=E2=80=99t=20download=20clang-format?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

---
 CMakeLists.txt | 37 -------------------------------------
 1 file changed, 37 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d087720..c0432d4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -282,43 +282,6 @@ endif()
 # make_readme.py script.
 file(COPY "${CEF_ROOT}/README.txt" DESTINATION "${CMAKE_BINARY_DIR}")
 
-# Download clang-format from Google Storage.
-if(OS_WINDOWS)
-  set(GS_PLATFORM "win32")
-  set(GS_HASHPATH "win/clang-format.exe.sha1")
-  set(GS_OUTPATH "win/clang-format.exe")
-elseif(OS_MACOSX)
-  set(GS_PLATFORM "darwin")
-  if("${CMAKE_HOST_SYSTEM_PROCESSOR}" STREQUAL "arm64")
-    set(GS_HASHPATH "mac/clang-format.arm64.sha1")
-  else()
-    set(GS_HASHPATH "mac/clang-format.x64.sha1")
-  endif()
-  set(GS_OUTPATH "mac/clang-format")
-elseif(OS_LINUX)
-  set(GS_PLATFORM "linux*")
-  set(GS_HASHPATH "linux64/clang-format.sha1")
-  set(GS_OUTPATH "linux64/clang-format")
-endif()
-
-message(STATUS "Downloading clang-format from Google Storage...")
-execute_process(
-  COMMAND "${PYTHON_EXECUTABLE}"
-          "tools/buildtools/download_from_google_storage.py"
-          "--no_resume"
-          "--platform=${GS_PLATFORM}"
-          "--no_auth"
-          "--bucket" "chromium-clang-format"
-          "-s" "tools/buildtools/${GS_HASHPATH}"
-          "-o" "tools/buildtools/${GS_OUTPATH}"
-  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-  RESULT_VARIABLE EXECUTE_RV
-  )
-if(NOT EXECUTE_RV STREQUAL "0")
-  message(FATAL_ERROR "Execution failed with unexpected result: ${EXECUTE_RV}")
-endif()
-
-
 #
 # Include target subdirectories.
 #
-- 
2.41.0

