--- a/bin/idea.sh	2021-04-06 13:18:41.000000000 +0300
+++ b/bin/idea.sh	2021-04-07 19:53:24.151450642 +0300
@@ -8,28 +8,11 @@
 message()
 {
   TITLE="Cannot start IntelliJ IDEA"
-  if [ -n "$(command -v zenity)" ]; then
-    zenity --error --title="$TITLE" --text="$1" --no-wrap
-  elif [ -n "$(command -v kdialog)" ]; then
-    kdialog --error "$1" --title "$TITLE"
-  elif [ -n "$(command -v notify-send)" ]; then
-    notify-send "ERROR: $TITLE" "$1"
-  elif [ -n "$(command -v xmessage)" ]; then
-    xmessage -center "ERROR: $TITLE: $1"
-  else
-    printf "ERROR: %s\n%s\n" "$TITLE" "$1"
-  fi
+  @NOTIFY_SEND@ "ERROR: $TITLE" "$1"
 }
 
-if [ -z "$(command -v uname)" ] || [ -z "$(command -v realpath)" ] || [ -z "$(command -v dirname)" ] || [ -z "$(command -v cat)" ] || \
-   [ -z "$(command -v egrep)" ]; then
-  TOOLS_MSG="Required tools are missing:"
-  for tool in uname realpath egrep dirname cat ; do
-     test -z "$(command -v $tool)" && TOOLS_MSG="$TOOLS_MSG $tool"
-  done
-  message "$TOOLS_MSG (SHELL=$SHELL PATH=$PATH)"
-  exit 1
-fi
+PATH='@PATH@'
+NATIVE_LIBRARY_PATH='@NATIVE_LIBRARY_PATH@'
 
 # shellcheck disable=SC2034
 GREP_OPTIONS=''
@@ -168,8 +151,9 @@
 # ---------------------------------------------------------------------
 IFS="$(printf '\n\t')"
 # shellcheck disable=SC2086
-"$JAVA_BIN" \
+exec "$JAVA_BIN" \
   -classpath "$CLASSPATH" \
+  -Djna.library.path="${NATIVE_LIBRARY_PATH}" \
   ${VM_OPTIONS} \
   "-XX:ErrorFile=$HOME/java_error_in_idea_%p.log" \
   "-XX:HeapDumpPath=$HOME/java_error_in_idea_.hprof" \
